app/gui/view/graph-editor/src/component/node.rs

```diff
@@ -654,6 +654,34 @@ impl NodeModel {
             self.display_object.add_child(&self.error_indicator);
         }
     }
+
+    /// Return the `SinglePortView` of the first output port of the node.
+    ///
+    /// Returns `None`:
+    /// 1. If there are no output ports.
+    /// 2. If the port does not have a `PortShapeView`.
+    /// 3. If the output port is [`MultiPortView`].
+    pub fn output_port(&self) -> Option<output::port::SinglePortView> {
+        let ports = self.output.model.ports();
+        let port = ports.first()?;
+        let shape = port.shape.as_ref()?;
+        use output::port::PortShapeView::Single;
+        match shape {
+            Single(shape) => Some(shape.clone_ref()),
+            _ => None,
+        }
+    }
```

Comment:
I don't like this function because it does not work with multi-output ports. I would be OK if such a function was defined in test-only module, but not in a normal, public API. Nodes should work correctly having multiple output ports, so either make this function returning all ports, or move it to a local, test-only module.

Also, The documentation states that it returns `None` if "If the port does not have a `PortShapeView`". This is not understandable from the perspective of the user of this API. When port could not have `PortShapeView`? Is this a situation that can normally happen? If so, when? (the same applies to `input_port` fn.